src/lib.rs

```diff
@@ -571,18 +616,15 @@ pub fn init(options: &InitOptions) -> Result<(ZWaveManager, mpsc::Receiver<ZWave
     let (mut zwave_manager, rx) = ZWaveManager::new(manager);
     try!(zwave_manager.add_watcher());
 
-    let device = match options.device {
-        Some(ref device) => device as &str,
-        _ => try!(get_default_device())
-    };
-
-    try!(
-        fs::File::open(device).map_err(|err| Error::CannotReadDevice(device.to_string(), err))
-    );
-
-    //println!("found device {}", device);
+    let devices = options.devices.clone().unwrap_or(get_default_devices());
```

Comment:
are you sure we need to clone `options.devices` ? Because with `iter` below we should get references only and so we don't need to clone ?